src/Invictus.Testing/LogicAppClient.cs

```diff
@@ -0,0 +1,498 @@
+            object originalAppDefinition = workflow.Definition;
+
+            _logger.LogTrace("Updates (+) the logic app '{LogicAppName}' workflow definition in resource group '{ResourceGroup}'", _logicAppName, _resourceGroup);
+            await UpdateAsync(workflow, JObject.Parse(logicAppDefinition));
+            return AsyncDisposable.Create(async () =>
+            {
+                _logger.LogTrace("Reverts (-) the update of the logic app '{LogicAppName}' workflow definition in resource group '{ResourceGroup}'", _logicAppName, _resourceGroup);
+                await UpdateAsync(workflow, originalAppDefinition);
+            });
+        }
+
+        private async Task UpdateAsync(Workflow workflow, object logicAppDefinition)
+        {
+            workflow.Definition = logicAppDefinition;
+            Workflow resultWorkflow = 
+                await _logicManagementClient.Workflows.CreateOrUpdateAsync(_resourceGroup, _logicAppName, workflow);
+
+            if (resultWorkflow.Name != _logicAppName)
+            {
+                throw new InvalidOperationException(
+                    $"Could not update the logic app '{_logicAppName}' workflow in resource group '{_resourceGroup}'correctly, "
+                    + "please make sure the authentication on this resource is set correctly and has the right access to this resource");
+            }
+        }
+
+        /// <summary>
+        /// Deletes the current logic app resource on Azure.
+        /// </summary>
+        public async Task DeleteAsync()
+        {
+            _logger.LogTrace("Deletes the workflow of logic app '{LogicAppName}' in resource group '{ResourceGroup}'", _logicAppName, _resourceGroup);
+            await _logicManagementClient.Workflows.DeleteAsync(_resourceGroup, _logicAppName);
+        }
+
+        /// <summary>
+        /// Runs the current logic app resource by searching for triggers on the logic app.
+        /// </summary>
+        /// <exception cref="LogicAppTriggerNotFoundException">When no trigger can be found on the logic app.</exception>
+        public async Task RunAsync()
+        {
+            string triggerName = await GetTriggerNameAsync();
+            await RunByNameAsync(triggerName);
+        }
+
+        /// <summary>
+        /// Runs the current logic app resource using the given <paramref name="triggerName"/>.
+        /// </summary>
+        /// <param name="triggerName">The name of the trigger that executes a workflow in the logic app.</param>
+        public async Task RunByNameAsync(string triggerName)
+        {
+            Guard.NotNullOrEmpty(triggerName, nameof(triggerName));
+
+            _logger.LogTrace("Run the workflow trigger of logic app '{LogicAppName}' in resource group '{ResourceGroup}'", _logicAppName, _resourceGroup);
+            await _logicManagementClient.WorkflowTriggers.RunAsync(_resourceGroup, _logicAppName, triggerName);
```

Comment:
This returns an object, does it include useful information we should surface such as a run id?

Reply:
Hmm, I don't know and when I test it, I got a `BadRequest`, is there anything I should know about this operation? Something that has to be in place before I can call this 'run' method? Not sure honestly what's the difference between 'run' and 'trigger' a logic app here. 🤔

Reply:
It's an exception that's being thrown. In both logic app's here in the test suite. 1) saying that the workflow trigger is disabled (while the logic app itself is enabled) and the 2) that the client tracking ID is missing.

Reply:
Then we should log that for sure :)

Sounds like a test going wrong, no? Can you help here @mbraekman ?

Reply:
Hmm, give me some time to look into this.
Feel free to "poke" me if I'm take too much time ;)

Reply:
If you're getting the following exception, that means the logic app really is still disabled, or at least according to this API.
```
"error": {
    "code": "WorkflowTriggerIsNotEnabled",
    "message": "Could not execute workflow 'TestLogicApp' trigger 'manual' with state 'Disabled': trigger is not enabled."
  }
```

Regarding the following exception, this is related to the setup of the Logic App in itself:
```
"error": {
    "code": "InvalidClientTrackingId",
    "message": "The 'clientTrackingId' value is not valid. The value cannot be null or all whitespace characters."
  }
```

The Logic App is expecting the trigger to contain a header "correlationId", which is being assigned to the clientTrackingId-property, which in turn is used to perform a `GetLogicAppRunAsync`.

Perhaps to test this specific operation, we'll create a separate logic app, which doesn't have those specific expectations -> this could be the same Logic App as I still need to create to test the "`UpdateLogicApp`"-method.

Reply:
We'll leave it up to that issue then to verify this return value.